src/status_im/ui2/screens/chat/components/chat_item.cljs

```diff
@@ -0,0 +1,199 @@
+                      (add-parsed-to-subheader acc-paragraph parsed-child)))
+                  {:components [rn/text]
+                   :length     0}
+                  children)
+
+                 "mention"
+                 {:components [rn/text @(re-frame/subscribe [:contacts/contact-name-by-identity literal])]
+                  :length     4} ;; we can't predict name length so take the smallest possible
+
+                 "status-tag"
+                 (truncate-literal (str "#" literal))
+
+                 "link"
+                 (truncate-literal destination)
+
+                 (truncate-literal literal))]
+    {:components (conj (:components acc) (:components result))
+     :length     (+ (:length acc) (:length result))}))
+
+(defn render-subheader
+  "Render the preview of a last message to a maximum of max-subheader-length characters"
+  [parsed-text]
+  (let [result
+        (reduce
+         (fn [{:keys [_ length] :as acc-text} new-text-chunk]
+           (if (>= length max-subheader-length)
+             (reduced acc-text)
+             (add-parsed-to-subheader acc-text new-text-chunk)))
+         {:components [rn/text {:style               styles/last-message-text
+                                :number-of-lines     1
+                                :ellipsize-mode      :tail
+                                :accessibility-label :chat-message-text}]
+          :length     0}
+         parsed-text)]
+    (:components result)))
+
+(defn content-type-community-invite? [content-type community-id]
+  (and (= constants/content-type-community content-type)
+       (not (string/blank? community-id))))
+
+(defn message-content-text [{:keys [content content-type community-id]}]
+  [rn/view
+   (cond
+     (not (and content content-type))
+     [preview-label :t/no-messages]
+
+     (and (or (= constants/content-type-text content-type)
+              (= constants/content-type-contact-request content-type)
+              (= constants/content-type-emoji content-type)
+              (= constants/content-type-command content-type))
+          (not (string/blank? (:text content))))
+     (if (string/blank? (:parsed-text content))
+       [rn/text {:style               styles/last-message-text
+                 :number-of-lines     1
+                 :ellipsize-mode      :tail
+                 :accessibility-label :chat-message-text}
+        (:text content)]
+       [render-subheader (:parsed-text content)])
+
+     (= constants/content-type-sticker content-type)
+     [preview-label :t/sticker]
+
+     (= constants/content-type-image content-type)
+     [preview-label :t/image]
+
+     (= constants/content-type-audio content-type)
+     [preview-label :t/audio]
+
+     (content-type-community-invite? content-type community-id)
+     (let [{:keys [name]}
+           @(re-frame/subscribe [:communities/community community-id])]
+       [preview-label :t/community-message-preview {:community-name name}]))])
+
+(def memo-timestamp
```

Comment:
Is the memoization really needed? A single millisecond difference is a cache miss.